diff --git forkSrcPrefix/svg-backend-nanosvg.c forkDstPrefix/svg-backend-nanosvg.c
index 154e5f5b3d889717dd1b561e3d87cedda4d84651..0292d0ac4a29b3866a44645e18476d96590e6bab 100644
--- forkSrcPrefix/svg-backend-nanosvg.c
+++ forkDstPrefix/svg-backend-nanosvg.c
@@ -7,9 +7,15 @@
 
 #define LOG_MODULE "fcft/svg"
 #define LOG_ENABLE_DBG 0
-#include "log.h"
-#include <nanosvg.h>
-#include <nanosvgrast.h>
+//#include <nanosvg.h>
+//#include <nanosvgrast.h>
+//#include "log.h"
+#define LOG_DBG(...)
+#define LOG_ERR(...)
+#define LOG_WARN(...)
+#define bool int
+#define false (0)
+#define true (1)
 
 #define min(x, y) ((x) < (y) ? (x) : (y))
 #define max(x, y) ((x) > (y) ? (x) : (y))
